Support DMIC sample and test on mcxn9xx boards

TomasBarakNXP opened the pull request:
Add dmic_api test support for mcx_n9xx_evk and frdm_mcxn947 boards.
Add Kconfig configuration of pcm width to dmic sample.
Add dmic sample support for mcx_n9xx_evk and frdm_mcxn947 boards.
Depends on #96980

TomasBarakNXP committed:
samples: dmic: enable on mcxn9xx boards

- add overlays for frdm_mcxn947 and mcxn_n9xx_evk boards in order to
  enable micfil for the dmic sample

Signed-off-by: Tomas Barak <tomas.barak@nxp.com>

TomasBarakNXP committed:
samples: dmic: Add sample bit width configuration

- in order to support different bit widths introduce a Kconfig parameter
  SAMPLE_BIT_WIDTH used in the main.c - default is 16 bits
- for platforms with NXP micfil peripheral, default is 32 bits
- BYTES_PER_SAMPLE needs to correspond to BIT_WIDTH

Signed-off-by: Tomas Barak <tomas.barak@nxp.com>

JarmouniA commented on samples/drivers/audio/dmic/boards/frdm_mcxn947_mcxn947_cpu0.conf:
```diff
@@ -0,0 +1,4 @@
+# Copyright (c) 2025 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_SAMPLE_BIT_WIDTH=32
```
instead of a adding a conf file for each board with the same value, you can do this
```
config SAMPLE_BIT_WIDTH
	int
	default 32 if SOC_SERIES_...
```

TomasBarakNXP committed:
tests: dmic_api: configure the parameters with Kconfig

- add Kconfig parameters for sample bit width and number of channels
- remove the conditional compilation from main.c to support the Kconfig
  configuration
- BYTES_PER_SAMPLE need to correspond to BIT_WIDTH

Signed-off-by: Tomas Barak <tomas.barak@nxp.com>

TomasBarakNXP committed:
tests: dmic_api: support mcxn9xx boards

- add mcxn9xx boards overlay to support 2 channels tests

Signed-off-by: Tomas Barak <tomas.barak@nxp.com>

ZhaoxiangJin reviewed (changes requested):
some minor fix need to apply

JarmouniA commented on tests/drivers/audio/dmic_api/src/main.c:
```diff
@@ -15,21 +15,11 @@
-#define PDM_CHANNELS 4 /* Two L/R pairs of channels */
-#define SAMPLE_BIT_WIDTH 16
+#define SAMPLE_BIT_WIDTH CONFIG_SAMPLE_BIT_WIDTH
+#define PDM_CHANNELS     CONFIG_PDM_CHANNELS
```
```suggestion
#define PDM_CHANNELS     CONFIG_SAMPLE_PDM_CHANNELS
```

TomasBarakNXP commented:
Requested changes applied.
